packages/react-instantsearch/src/components/SearchBox.js

```diff
@@ -120,6 +120,7 @@ class SearchBox extends Component {
   onSubmit = e => {
     e.preventDefault();
     e.stopPropagation();
+    this.input.blur();
```

Comment:
Same behaviour on Google and Facebook, submit load the search and focusout. should be safe enough